pandas/core/arrays/arrow/array.py

```diff
@@ -726,6 +727,13 @@ def _evaluate_op_method(self, other, op, arrow_funcs):
         result = pc_func(self._pa_array, other)
         return type(self)(result)
 
+    @staticmethod
+    def _evaluate_binary_repeat(binary, integral):
+        if not pa.types.is_integer(integral.type):
```

Comment:
Probably this can be removed and let pyarrow bubble up a NotImplementedError.

Reply:
Since you're already checking this in the new `elif` branch, you can just add this to:

```python
elif op in [operator.mul, roperator.rmul] and pa.types.is_integer(integral.type):
```

Reply:
Good point.